core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcServerManager.java

```diff
@@ -99,10 +98,11 @@
                                                        .build();
     private final ExecutorService executor = Executors.newCachedThreadPool(threadFactory);
     private Map<RpcModule<RpcRequest, RpcResponse>, KafkaConsumerRunner> rpcModuleConsumers = new ConcurrentHashMap<>();
-    // cache to hold rpcId for directed RPCs and expire them
-    private Cache<String, Long>  rpcIdCache;
     // cache to hold rpcId and ByteString when there are multiple chunks for the message.
-    private Cache<String, ByteString> messageCache;
+    private Map<String, ByteString> messageCache = new ConcurrentHashMap<>();
```

Comment:
We should be removing the messages from this map when the corresponding RPC request has expired - i.e. if we have received some but not all all of the chunks before the TTL.

Reply:
Not receiving all chunks can only happen if  OpenNMS is restarting and minion is still alive. Updated code to handle this as well.